### PR TITLE
[#162] Implement comment component

### LINKED ISSUE #162: Refactor comment component
**Is your feature request related to a problem? Please describe.**
Need to investigate the existing code in place to provide commenting functionality on posts and ultimately get comments working

**Describe the solution you'd like**
Be able to make the changes required (on the frontend) to comment on posts 

**Describe alternatives you've considered**
N/A

**Additional context**
This will require both frontend and backend changes as I don't believe commenting is currently implemented on the backend. This issue relates to the frontend

### PR DESCRIPTION
**Pull request by sv-s-dev**
# Description:
Fixes #162 

- Implemented a comment component that pops up when a user clicks the comment button on a post
- Shows tags and handles properly using PR #189 
- Uses post input and multiple other components in the modal
- submitForm() function in CommentController still needs to be implemented (waiting on backend) for posting the comment to API

image

### All Submissions:

* [x] Have you followed the guidelines in our [Contributing document](../CONTRIBUTING.md)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../../pulls) for the same update/change?

### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] My code follows the code style of this project
- [x] My code has been commented
- [ ] My change requires a change to the documentation
- [ ] I have updated the documentation accordingly
- [ ] I have added tests to cover my changes
- [x] All new and existing tests passed

### CODE REVIEW

**Comment by KirstyGong** [comment]
I think the component would look better if you can make the "Replying to ___" text and the corresponding input box closer together

**Review by KirstyGong — APPROVED** [review]
LGTM

**Review by ppat965 — DISMISSED** [review]
I feel the MVC is not followed completely. Eg. styling is in Comment View rather then only in comment.module.scss.

**Review by dogeliness — APPROVED** [review]
Looks good, I think it just wants you to clean up some of the console.log statements

File: frontend/src/components/posts/interactions/InteractionsController.jsx [hunk 1]
```diff
@@ -1,29 +1,40 @@
...
     // todo, implement like business logic
   }
 
   const onComment = () => {
-    // todo, open a reply page
+    if (!showComponent) {
+      setShowComponent(true)
+    } else {
+      setShowComponent(false)
+    }
```
Comment (scorpionknifes) [inline frontend/src/components/posts/interactions/InteractionsController.jsx]:
```suggestion
    setShowComponent(!showComponent)
```

File: frontend/src/components/posts/comment/CommentView.jsx [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+              __html: processMentions({
+                content: `Replying to @${postData.author.username} ` ?? '',
+                tags,
+                handles,
+                tagStyle: styles.tagStyle,
+                handleStyle: styles.handleStyle,
+              }),
+            }}
+            style={{ marginTop: 25 }}
+          />
```
Comment (scorpionknifes) [inline frontend/src/components/posts/comment/CommentView.jsx]: Can you style this using a 
```html
<p>Replying to <a href="/user/???"">@Author</a></p>
```

The processMention runs an intensive loop would be better if we don't use it for something we can hard code.

**Review by scorpionknifes — CHANGES_REQUESTED** [review]
LGTM, just a couple changes

**Comment by sv-singh** [comment]
> I think the component would look better if you can make the "Replying to ___" text and the corresponding input box closer together

Sure I made them a bit closer now

**Comment by sv-s-dev** [comment]
> I think the component would look better if you can make the "Replying to ___" text and the corresponding input box closer together

Sure I made them a bit closer now

**Comment by sv-s-dev** [comment]
> I feel the MVC is not followed completely. Eg. styling is in Comment View rather then only in comment.module.scss.

Yep fixed this

File: frontend/src/components/posts/comment/CommentView.jsx (hunk 2 above)
Comment (sv-s-dev) [reply frontend/src/components/posts/comment/CommentView.jsx]: Cool I've made those changes

**Review by scorpionknifes — APPROVED** [review]
LGTM

**Pull request merged**